internal/mkcgo/mkcgo.go

```diff
@@ -26,18 +32,63 @@ type Enum struct {
 	Value string
 }
 
+// FuncAttrs contains attributes of a function.
+type FuncAttrs struct {
+	Tags           []TagAttr
+	VariadicTarget string
+	Optional       bool
+	NoError        bool
+	ErrCond        string
+	NoEscape       bool
+	NoCallback     bool
+}
+
 // Func describes a function.
 type Func struct {
-	FuncAttributes
+	FuncAttrs
 	Name   string
 	Params []*Param
-	Ret    *Return
+	Ret    string
 }
 
+// Variadic returns true if the ends with a variadic parameter.
 func (f *Func) Variadic() bool {
 	return len(f.Params) > 0 && f.Params[len(f.Params)-1].Variadic()
 }
 
+// ImportName returns the import name of the function.
+func (f *Func) ImportName() string {
+	if f.VariadicTarget != "" {
+		return f.VariadicTarget
+	}
+	return f.Name
+}
+
+// String returns a string representation of the function,
+// which is not necessarily valid Go nor C code.
+func (f *Func) String() string {
+	var bld strings.Builder
+	if f.Ret != "" {
+		bld.WriteString(f.Ret)
+		bld.WriteByte(' ')
+	}
+	bld.WriteString(f.Name)
+	bld.WriteByte('(')
+	for i, p := range f.Params {
+		if i > 0 {
+			bld.WriteString(", ")
+		}
+		bld.WriteString(p.Type)
+		if p.Name != "" {
+			bld.WriteByte(' ')
+			bld.WriteString(p.Name)
+		}
+	}
+	bld.WriteString(") ")
+	bld.WriteString(fmt.Sprintf("%v", f.FuncAttrs))
```

Comment:
IMO the result of this is puzzling... `int TestFunc() {[]  false false  false false}` in the test case. Maybe `FuncAttrs` should have a `String()` too, so it can output nothing in this case, and so it doesn't rely on positional bools in other cases?